File: dev.js
const os = require("os");
const { exec } = require("child_process");

WINDOWS_COMMAND = "npm run build-dev && cd build\\Release && preview.exe";
UNIX_COMMAND = "npm run build-dev && cd build/ && ./preview ";

const watch = () => {
  exec("npm run watch", (error, stdout, stderr) => {
    if (error) {
      console.log(`BUILD ERROR: ${error.message}`);
    }
    if (stderr) {
      console.log(`STDERR: ${stderr}`);
      return;
    }
    console.log(`STDOUT: ${stdout}`);
  });
};

switch (os.platform()) {
  case "linux" || "darwin":
    exec(UNIX_COMMAND, (error, stdout, stderr) => {
      if (error) {
        console.log(`BUILD ERROR: ${error.message}`);
      }
      if (stderr) {
        console.log(`STDERR: ${stderr}`);
        return;
      }
      console.log(`STDOUT: ${stdout}`);
    });
    console.log("DEV BUILD SUCCESSFULL");
    watch();
    break;

  case "win32":
    exec(WINDOWS_COMMAND, (error, stdout, stderr) => {
      if (error) {
        console.log(`BUILD ERROR: ${error.message}`);
      }
      if (stderr) {
        console.log(`STDERR: ${stderr}`);
        return;
      }
      console.log(`STDOUT: ${stdout}`);
    });
    console.log("DEV BUILD SUCCESSFULL");
    watch();
    break;
}

File: dev.py
import os
from threading import Thread


OS = os.name    # Windows - "nt"  ,  Mac/Linux - "posix"

# POSIX cd build/ && ./preview
# Windows cd build\\Release && preview.exe
POSIXCMD = "cd build/ && ./preview"
NTCMD = "cd build\\Release && preview.exe"

NPMCMD = "npm start"


def executeCommand(command: str):
    try:

        os.system(command)
    except Exception as e:
        print("ERROR IN executeCommand():", e)


def openDev(system):
    try:

        npm_thread = Thread(target=executeCommand, args=[NPMCMD])

        if system == "nt":
            preview_thread = Thread(target=executeCommand, args=[NTCMD])
        else:
            preview_thread = Thread(target=executeCommand, args=[POSIXCMD])

        npm_thread.start()
        preview_thread.start()

        npm_thread.join()
        preview_thread.join()

    except Exception as e:
        print("ERROR IN openDev():", e)


if __name__ == "__main__":

    openDev(OS)
